File: 387-FirstUniqueCharacterinaString/387-FirstUniqueCharacterinaString.cpp
class Solution {
public:
    int firstUniqChar(string s)
    {
        unsigned int count[26] = { 0 };

        for (const auto & c : s)
            count[c - 'a']++;

        int ans = 0;
        for (const auto & c : s)
            if (count[c - 'a'] == 1) return ans;
            else ans++;

        return -1;
    }
};

File: 387-FirstUniqueCharacterinaString/387-FirstUniqueCharacterinaString.kt
class Solution {
    fun firstUniqChar(s: String): Int
    {
        val counts = IntArray(26)

        for (c in s) {
            counts[c - 'a']++
        }

        var i = 0;
        for (c in s) {
            if (counts[c - 'a'] == 1) return i
            else i++
        }
        

        return -1
    }
}